python/ray/worker.py

```diff
@@ -1524,16 +1525,6 @@ def put(value, weakref=False):
                     "ray.put(value, weakref=True) to allow object data to "
                     "be evicted early.")
                 raise
-        # Pin the object buffer with the returned id. This avoids put returns
-        # from getting evicted out from under the id.
-        # TODO(edoakes): we should be able to avoid this extra IPC by holding
-        # a reference to the buffer created when putting the object, but the
-        # buffer returned by the plasma store create method doesn't prevent
-        # the object from being evicted.
-        if not weakref and not worker.mode == LOCAL_MODE:
-            object_id.set_buffer_ref(
-                worker.core_worker.get_objects([object_id],
-                                               worker.current_task_id))
```

Comment:
I think it's easier to keep this code and just let the owner to pin the objects in plasma store, instead of going through raylet.

Reply:
Ah yes that would be simpler, but it only works when the worker is on the same node as the objects that it owns. If a worker spawns a task that creates a remote plasma object, it is the owner of that object and needs to pin it but we don't want to have to transfer a copy of all owned plasma objects to the same node as the owner.

Reply:
Yes that's true. I meant that if worker is on the same then it's simpler and cheaper for the owner to pin it. In other cases we need raylet to do it. The drawback is we need two different solutions to handle different cases, but this is probably ok. What do you think?

Reply:
Ah I see your point now. We also discussed this but feel that it would be better to focus on simplicity and correctness for now by having a single code path and optimize later. Given that this code is only exercised for large objects, it's also unclear if these extra RPCs are a performance bottleneck.